sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileDataSource.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.binaryfile
+
+import org.apache.spark.sql.types._
+
+/**
+ * `binaryfile` package implements Spark SQL data source API for loading binary file data
+ * as `DataFrame`.
+ *
+ * The loaded `DataFrame` has two columns, the schema is:
+ *  - status: `StructType` (the file status information)
+ *  - content: `BinaryType` (binary data of the file content)
+ *
+ * The schema of "status" column described above is:
+ *  - path: `StringType` (the file path)
+ *  - modification_time: `TimestampType` (last modification time of the file)
```

Comment:
It might be useful to mention on some FS implementation, this might not available and fallback to some default value.